File: apps/api/app/models/profile_verification.py
from datetime import datetime, timezone
from uuid import UUID, uuid4

from sqlalchemy import Boolean, DateTime, ForeignKey, String, Uuid
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class ProfileVerification(Base):
    __tablename__ = "profile_verifications"

    id: Mapped[UUID] = mapped_column(
        Uuid,
        primary_key=True,
        default=uuid4,
    )
    user_id: Mapped[UUID] = mapped_column(
        Uuid,
        ForeignKey("users.id", ondelete="CASCADE"),
        unique=True,
        nullable=False,
        index=True,
    )
    linkedin_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    website_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    company_registration: Mapped[str | None] = mapped_column(String(255), nullable=True)
    is_verified: Mapped[bool] = mapped_column(
        Boolean,
        default=False,
        server_default="0",
        nullable=False,
    )
    rejection_reason: Mapped[str | None] = mapped_column(String(500), nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=lambda: datetime.now(timezone.utc),
        nullable=False,
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=lambda: datetime.now(timezone.utc),
        onupdate=lambda: datetime.now(timezone.utc),
        nullable=False,
    )

    @property
    def badges(self) -> list[str]:
        badges: list[str] = []
        if self.is_verified:
            badges.append("Verified Profile")
        if self.linkedin_url:
            badges.append("LinkedIn submitted")
        if self.website_url:
            badges.append("Website submitted")
        if self.company_registration:
            badges.append("Company registration submitted")
        return badges

File: apps/web/src/lib/api/profile-verification.ts
import { apiClient } from "@/lib/api/client";

export type VerificationBadge =
  | "LinkedIn submitted"
  | "Website submitted"
  | "Company registration submitted";

export type ProfileVerification = {
  id: string;
  user_id: string;
  linkedin_url: string | null;
  website_url: string | null;
  company_registration: string | null;
  verification_badges: VerificationBadge[];
  created_at: string;
  updated_at: string;
};

export type ProfileVerificationPayload = {
  linkedin_url?: string;
  website_url?: string;
  company_registration?: string;
};

export async function getMyProfileVerification() {
  const response = await apiClient.get<ProfileVerification>("/profile-verification/me");
  return response.data;
}

export async function saveMyProfileVerification(payload: ProfileVerificationPayload) {
  const response = await apiClient.put<ProfileVerification>(
    "/profile-verification/me",
    payload
  );
  return response.data;
}
